op-acceptance-tests/tests/flashblocks/flashblocks_stream_test.go

```diff
@@ -0,0 +1,199 @@
+)
+
+// TestFlashblocksStream checks we can connect to the flashblocks stream
+func TestFlashblocksStream(gt *testing.T) {
+	t := devtest.SerialT(gt)
+	sys := presets.NewSimpleFlashblocks(t)
+	logger := testlog.Logger(t, log.LevelInfo).With("Test", "TestFlashblocksStream")
+	tracer := t.Tracer()
+	ctx := t.Ctx()
+	logger.Info("Started Flashblocks Stream test")
+
+	ctx, span := tracer.Start(ctx, "test chains")
+	defer span.End()
+
+	ctx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+
+	// Test all L2 chains in the system
+	for l2Chain, flashblocksBuilderSet := range sys.FlashblocksBuilderSets {
+		_, span = tracer.Start(ctx, "test chain")
+		defer span.End()
+
+		networkName := l2Chain.String()
+		t.Run(fmt.Sprintf("L2_Chain_%s", networkName), func(tt devtest.T) {
+			expectedChainID := l2Chain.ChainID().ToBig()
+			for _, flashblocksBuilderNode := range flashblocksBuilderSet {
+				flashblocksBuilderNodeStack := flashblocksBuilderNode.Escape()
+				require.Equal(t, flashblocksBuilderNodeStack.ChainID().ToBig(), expectedChainID, "flashblocks builder node chain id should match expected chain id")
+
+				var associatedConductor stack.Conductor
+				for _, conductor := range sys.ConductorSets[l2Chain] {
+					if flashblocksBuilderNodeStack.ConductorID() == conductor.Escape().ID() {
+						associatedConductor = conductor.Escape()
+					}
+				}
+				require.NotNil(t, associatedConductor, "there must be a conductor associated with the flashblocks builder node")
+
+				mode := FlashblocksStreamMode_Follower
+				if dsl.NewConductor(associatedConductor).IsLeader() {
+					mode = FlashblocksStreamMode_Leader
+				}
+
+				testFlashblocksStream(tt, logger, dsl.NewFlashblocksBuilderNode(flashblocksBuilderNodeStack), mode)
+			}
+		})
+	}
+}
+
+// testFlashblocksStream tests the presence / absence of a flashblocks stream operating at a 250ms rate
+func testFlashblocksStream(t devtest.T, logger log.Logger, flashblocksBuilderNode *dsl.FlashblocksBuilderNode, mode FlashblocksStreamMode) {
+	t.Run(fmt.Sprintf("Flashblocks_Stream_%s", mode), func(t devtest.T) {
+		require.Contains(t, []FlashblocksStreamMode{FlashblocksStreamMode_Leader, FlashblocksStreamMode_Follower}, mode, "mode should be either leader or follower")
+		require.NotNil(t, flashblocksBuilderNode, "flashblocksBuilderNode should not be nil")
+
+		wsURL := flashblocksBuilderNode.Escape().FlashblocksWsUrl()
+		logger.Info("Testing WebSocket connection to", "url", wsURL)
+
+		dialer := &websocket.Dialer{
+			HandshakeTimeout: 6 * time.Second,
+		}
+
+		conn, _, err := dialer.Dial(wsURL, nil)
+		require.NoError(t, err, "failed to connect to FLashblocks WebSocket endpoint", "url", wsURL)
+		defer conn.Close()
+
+		logger.Info("WebSocket connection established, reading stream for 5 seconds")
+
+		streamedMessages := make([]string, 0)
+
+		timeout := time.After(5 * time.Second)
+		for {
+			select {
+			case <-timeout:
+				goto done
```

Comment:
nit: I find this more readable (just a suggestion)

```suggestion
ReadMessages:
		for {
			select {
			case <-timeout:
				break ReadMessages
```

(then the 'done:' label and goto aren't needed and I find it reads, a little, less jarringly)

Reply:
I usually find goto's more reabable lol. It's usually more suggestive of "where the next line of code is" as opposed to the break's nature "where the next line of code is not".

But I don't have any strong opinion so let me know if you'd still like to suggest `break` instead of `goto`